resources/assets/components/ReviewButton.js

```diff
@@ -26,6 +26,16 @@ const ReviewButton = ({ post, status, children }) => {
       id: post.id,
       status: status,
     },
+    // We'll optimistically update the interface with the given status
+    // before waiting for the full network round-trip. Snappy!
```

Comment:
Snappy!! If something goes wrong will it switch back to the status from the db?

Reply:
Yep! This "pretends" the given response was returned, and then replaces it with the real response once we get it. (So for things like this where they're almost certain to be the same, it's a pretty cool way to make it seem a ton faster.)

Reply:
THIS IS SO COOL!!!!!!